### ISSUE TITLE
How to locally save the external images from an API?

### ISSUE THREAD

**Issue by teddylau168** [react-static/react-static#1288]
I'm doing some research and making some demo using react-static.

I found a problem with the images, which is external images from the API instead of the images in CSS or local image for the UI.

For example, I've developed an API with Wordpress basic, there're a lot of posts are returned by that API, sometimes there are a lot of images are returned as "Array" like the following JSON.

```
{
  "title": "testing title...",
  "content": "testing content...",
  "images": [
    "http://testing.abc.com/images/upload/testing1.jpg",
    "http://testing.abc.com/images/upload/testing2.jpg",
    "http://testing.abc.com/images/upload/testing3.jpg"
  ]
}
```

Is there any way to download the image and put them into the the <img tag as a relative path instead of absolute path, when we build a static website. So the result should be ...
```
<img src="./public/images/upload/testing1.jpg"/>
or (based on the page url route)
<img src="../../public/images/upload/testing1.jpg"/>
```
instead of 
```
<img src="http://testing.abc.com/images/upload/testing1.jpg"/>
```

**Comment by Pixelatex** [comment]
You can use webpack configs to do this but why not just use the images from the wordpress server? Just cache them with cloudflare

**Comment by teddylau168** [comment]
Because I need to build a website totally static and offline. Which means even if I put them into a USB and browse it in the computer without network.

**Comment by Pixelatex** [comment]
Right... so thinking a bit on how you could do this flow in your static config.

1. Fetch the posts from your api.
2. Iterate over the posts
3. Fetch the image(s) for that post and stream the result to a folder (fssync or something).
4. Pass the resulting file path, .e.g. `images/file1.jpg` as a prop to the posts route.
5. Profit!

**Comment by teddylau168** [comment]
Yup, it’s what I done exactly

**Comment by Pixelatex** [comment]
Alternatively, if someone else is looking into this. You could convert the images as a base64 string and pass those along as props. This eliminates the need to save the files locally.

Note: this is only useful for offline sites as this prevents image caching and makes your files rather big for web with all the included base 64-strings.

**Comment by sytabaresa** [comment]
This process is a genuine candidate to be implemented in a plugin, related to #1130 issue, maybe a component that automatically download images in a _static_ directory depending of they use.
